migrator.php

```diff
@@ -1240,7 +1246,13 @@ private function create_or_update_woo_order( $shopify_order, $woo_order ) {
 		$this->process_order_refunds( $order, $shopify_order );
 	}
 
-	private function create_or_assign_customer( $order, $shopify_order ) {
+	private function create_or_assign_customer( $order, $shopify_order, $mode = 'test' ) {
+		// Mask customer email and remove phone number in test mode.
+		if ( $mode === 'test' ) {
+			$shopify_order->email = $shopify_order->email . '.masked';
+			$shopify_order->phone = '';
```

Comment:
We're actually using `$shopify_order->billing_address->phone` and `$shopify_order->shipping_address->phone` instead of the `$shopify_order->phone` to assign the phone number to the customer

Reply:
Thanks for spotting this! 
I was unable to test the phone numbers accurately until late last night, when I created an order with phone number in it, on the test site. I smelled at that time the phone number was not getting redacted. I later checked the [Shopify order](https://shopify.dev/docs/api/admin-rest/2023-10/resources/order#resource-object) structure ( should've done it earlier! ) .  I will fix this shortly.

Reply:
I have fixed this here - 7672b28

I have also moved the phone masking early on, since it is also required by the [process_order_addresses](https://github.com/woocommerce/migrator-cli/blob/7672b2838bebf3e5f1948ba0b6ee5825f2d12736/migrator.php#L1336) function. Email is not needed at this stage.

Reply:
I've done a test run now, and none of the orders have a phone number, only Nagesh Second has 9999999. I can see in the Shopify admin that other customers have phone number in the Shopify admin.

Reply:
@Ninodevo - Thanks for manually testing this. This had me puzzled for a while too. Until I checked the [Shopify order](https://shopify.dev/docs/api/admin-rest/2023-10/resources/order#resource-object) structure, and compared the way we read and store phone numbers. 

Shopify imports phone numbers only if its included in  `$shopify_order->billing_address->phone` or `$shopify_shipping_address`. 9999999 is inserted only if there is a number present in either of these fields. 

Other than these two, Shopify has a provision to store a phone number on the customer profile itself - `$shopify_order->customer->phone` .. The cases where you see phone numbers but not 999999, are those that have this, but do not have shipping nor billing phone in the order that is imported. (WordPress + Woo does not have a field for the customer itself, other than billing and shipping ) . I hope this makes sense.

Reply:
@Ninodevo , I have merged the changes for now, since most feedback has been addressed. Happy to reopen if needed.

Reply:
All good, thanks!